src/daemon/rpmostree-sysroot-upgrader.cxx

```diff
@@ -434,12 +434,19 @@ rpmostree_sysroot_upgrader_pull_base (RpmOstreeSysrootUpgrader *self, const char
           return glnx_throw (error, "Specifying commit overrides for container-image-reference "
                                     "type refspecs is not supported");
         if (check)
-          return glnx_throw (error, "Cannot currently check for updates without downloading");
+          {
+            *out_changed = FALSE;
```

Comment:
I think this is the change I feel least comfortable with.  I'm not saying it's wrong, but...it feels like we're bypassing the system here and just saying "if you're in a container then nothing changed"...and then we actually go to check for changes later.

So far, the logic in `rpmostreed_update_generate_variant()` has been intending to compute state from *cached* data, but that's where we're doing the check.

And then the problem here is, we're not caching the updated manifest anywhere.  I think it'd really help us though to switch to doing that.

Reply:
And to flesh that out a bit, I'd propose that we drive this "check for updates" API and flow into ostree-ext.

Reply:
Are you saying we close this PR and work on the above? Then rpm-ostree --update check  if in a container flow just calls ostree-ext and all the output comes from ostree-ext?

Reply:
I don't think we should close.  Perhaps what we can do here is serialize the fetched manifest into e.g. a file like `/run/rpm-ostree/pending-manifest.json` or so...which we'd need to take care of cleaning when things change, but not too hard.

Reply:
OK some WIP in https://github.com/ostreedev/ostree-rs-ext/issues/496#issue-1791085555